src/java/org/apache/cassandra/utils/btree/BTree.java

```diff
@@ -2569,30 +2575,31 @@ void copyNoOverflow(Object[] source, int offset, int length)
         }
 
         /**
-         * Copy the contents of {@code source[from..to)} to {@code buffer}, overflowing as necessary.
-         * Applies {@code updateF} to the contents before insertion.
+         * Copy the contents of the data to {@code buffer}, overflowing as necessary.
          */
-        <Insert, Existing> void copy(Object[] source, int offset, int length, UpdateFunction<Insert, Existing> updateF)
+        <Insert, Existing> void copy(Object[] source, int offset, int length, UpdateFunction<Insert, Existing> apply)
         {
-            if (isSimple(updateF))
+            if (isSimple(apply))
             {
                 copy(source, offset, length);
                 return;
             }
 
-            if (count + length > MAX_KEYS)
+            for (int i = 0; i < length; ++i)
```

Comment:
This change should no longer be necessary, and we should state in `update`s definition that it can't return nulls as other parts of `update` cannot handle them.